awx/settings/defaults.py

```diff
@@ -427,6 +427,7 @@ def IS_TESTING(argv=None):
     'gather_analytics': {'task': 'awx.main.tasks.gather_analytics', 'schedule': timedelta(minutes=5)},
     'task_manager': {'task': 'awx.main.scheduler.tasks.run_task_manager', 'schedule': timedelta(seconds=20), 'options': {'expires': 20}},
     'k8s_reaper': {'task': 'awx.main.tasks.awx_k8s_reaper', 'schedule': timedelta(seconds=60), 'options': {'expires': 50}},
+    'receptor_reaper': {'task': 'awx.main.tasks.awx_receptor_workunit_reaper', 'schedule': timedelta(seconds=60)},
```

Comment:
once per minute seems aggressive. Won't this run on every node in the cluster each time? I worry about scaling up to large numbers of nodes, like 30 nodes.

Reply:
I don't think we support that many control nodes. When we switch to having receptor execution nodes, this logic will need to be expanded then we can reassess. At 30 seconds, that matches the k8s reaper so I am fine with it.

Reply:
We won't need to change anything as we introduce purely execution nodes, will we?

My entire concern is that these receptorctl actions apply to the whole network. If you only ran _one_ instance of this task once per period that would accomplish the objective just fine (it's not some queue specific to this control node, if I understand). We just have no mechanism to have a whole-cluster periodic task.

Reply:
The scheduled jobs function + the analytics functionality are cluster-wide periodic tasks. We trigger a periodic beat per-node, use a lock to make sure the exec is exclusive, update a database timestamp to ensure only 1 execution during the desired period.

> We won't need to change anything as we introduce purely execution nodes, will we?
It's subtle, but we rely on local information from the dispatcher to rectify execution. What this means is that the reaper must run on each control node. More specifically, the reaper can only reap jobs that have the same control node that the reaper is running on. 
`receptorctl work list` must target a single receptor execution node. For a single control node, we will have to make multiple calls to `receptorctl work list` to get all of the `exexecution_node` work units.

We are also thinking about dropping the need for a single-specific control node "babysitter". That actually might simplify the reaping process. I'm unsure.

Reply:
okay, you're talking about a call `receptorctl work list --node=<node_id>` call. Yes, that makes it self-obvious that it's a per-node call.